lib/Entry.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Entry
+  ( dateModified
+  , Entry, EntryConvertible (..), newEntry
+  , path, masterField, fields
+  , Field (..), FieldKey (..)
```

Comment:
```hs
FieldKey (..)
```

The field `FieldKey` has some invariants (all characters are valid & has at least 1 character), so its default constructor `FieldKey :: Text -> FieldKey` should not be exported.

Reply:
@Heimdell Are there more beside `To/FromJSON`?